pkg/services/ngalert/state/state.go

```diff
@@ -177,16 +177,20 @@ type Evaluation struct {
 	// Values contains the RefID and value of reduce and math expressions.
 	// Classic conditions can have different values for the same RefID as they can include multiple conditions.
 	// For these, we use the index of the condition in addition RefID as the key e.g. "A0, A1, A2, etc.".
-	Values map[string]*float64
+	Values map[string]float64
```

Comment:
Makes these values consistent with the `state.Values`.

This has no behavioral impact - in the pointer case, we later convert pointers to NaN's anyway. Now we convert to NaN - when building this object the result is exactly the same. Added benefits of not having to deal with taking pointers to constants in tests :)

These values are completely duplicated with state.Values, we could just drop this entirely - future PR though, it makes the logic here even harder to follow.